### PR TITLE
multiple changes, CalibrationFile, CenterMatcher, EventAnalyzer

### PR DESCRIPTION
**Pull request by dorotheapfeiffer**
### Issue reference / description

The branch you merge from should already reference an event-formation-unit github ticket number. You can add a descriptive title, but if an issue is referenced, you don't have to.

## Checklist for submitter

- [x] Check for conflict with integration test
- [x] Unit tests pass

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [ ] Recommend for group code review

Also, nominate it on the code_review Slack channel.

### CODE REVIEW

**Comment by dorotheapfeiffer** [comment]
Huhu Morten,

here we go.

Cheers

Doro

File: prototype2/common/reduction/matching/AbstractMatcher.cpp [hunk 1]
```diff
@@ -9,8 +9,8 @@
 #include <common/reduction/matching/AbstractMatcher.h>
 #include <common/Trace.h>
 
-// #undef TRC_LEVEL
-// #define TRC_LEVEL TRC_L_DEB
+#undef TRC_LEVEL
```
Comment (mortenjc) [inline prototype2/common/reduction/matching/AbstractMatcher.cpp]: We should get rid of traces before comitting to master

File: prototype2/gdgem/NMXConfig.cpp [hunk 2]
```diff
@@ -50,8 +51,7 @@ NMXConfig::NMXConfig(std::string configfile, std::string calibrationfile) {
     root = nlohmann::json::parse(jsonstring);
   }
   catch (...) {
-    XTRACE(INIT, WAR, "Invalid Json file: %s", configfile.c_str());
-    return;
+    XTRACE(INIT, WAR, "Invalid Json file: %s", configfile.c_str());
```
Comment (mortenjc) [inline prototype2/gdgem/NMXConfig.cpp]: did you not want to throw a runtime exception in this case?

Comment (mortenjc) [reply prototype2/gdgem/NMXConfig.cpp]: Also I guess the severity is probably ERR, right?

File: prototype2/gdgem/NMXConfig.cpp [hunk 3]
```diff
@@ -114,38 +120,34 @@ NMXConfig::NMXConfig(std::string configfile, std::string calibrationfile) {
...
+      analyzer_ = std::make_shared<EventAnalyzer>(time_algorithm);
       // \todo deduce geometry from SRS mappings eventually
       //       this is better for now, while prototyping
       geometry.nx(root["geometry_x"].get<unsigned int>());
       geometry.ny(root["geometry_y"].get<unsigned int>());
       geometry.nz(1);
       geometry.np(1);
     }
-
+
```
Comment (mortenjc) [inline prototype2/gdgem/NMXConfig.cpp]: empty lines should not have whitespace in them :-)

**Comment by dorotheapfeiffer** [comment]
Yes, I think that it is best to have a run time exception, Without a valid json, there is no configuration, no mapping, so one can really not run in this case..

**Comment by dorotheapfeiffer** [comment]
How do I make the code again to comply with your standards, is there a tool ?

File: prototype2/gdgem/NMXConfig.cpp [hunk 4]
```diff
@@ -114,38 +120,34 @@ NMXConfig::NMXConfig(std::string configfile, std::string calibrationfile) {
...
-    {
-      auto MGA = std::make_shared<MGAnalyzer>();
-      MGA->weighted(root["analyze_weighted"].get<bool>());
-      geometry.nx(MGA->geometry().x_range());
-      geometry.ny(MGA->geometry().y_range());
-      geometry.nz(MGA->geometry().z_range());
-      geometry.np(1);
-
-      analyzer_ = MGA;
+    if (root.count("analyzer")) {
```
Comment (mortenjc) [inline prototype2/gdgem/NMXConfig.cpp]: Are you changing the json format? If so please update the json files in DM/data/EFU_reference/gdgem 
to reflect this. If not the integration tests will fail.

**Comment by dorotheapfeiffer** [comment]
The changes are equivalent to the ones Martin did, I always check for the presence of the parameter, e.g.

if (root.count("analyzer"))
if (root.count("clusterer"))
if (root.count("matcher"))

So if the new parameter is there, good, if not then it takes some default value. So old json files should still work.

But are there some test data sets for the MG VMM3 stuff ?

**Pull request merged**